liquidhaskell-boot/src/Language/Haskell/Liquid/Bare.hs

```diff
@@ -694,6 +709,12 @@ addReflSigs :: Bare.Env -> ModName -> BareRTEnv -> GhcSpecRefl -> GhcSpecSig ->
 ------------------------------------------------------------------------------------------
 addReflSigs env name rtEnv refl sig =
   sig { gsRefSigs = F.notracepp ("gsRefSigs for " ++ F.showpp name) $ map expandReflectedSignature reflSigs
+      -- We make sure that the reflected functions are excluded from `gsAsmSigs`, except for the signatures of
+      -- actual functions in assume-reflect. The latter are added here because 1. it's what makes tests work
+      -- and 2. so that we probably "shadow" the old signatures of the actual function correctly. Note that even if the
+      -- signature of the actual function was asserted and not assumed, we do not put our new signature for the actual function
+      -- in `gsTySigs` (which is for asserted signatures). Indeed, the new signature will *always* be an assumption since we
+      -- add the extra post-condition that the actual and pretended function behave in the same way.
```

Comment:
@nikivazou, does this ring well to you?

Reply:
Even though this is merged, I still would like to know what you think and would contribute edits if necessary.

Reply:
This makes sense to me!